src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api/Validators/SortStringValidators.cs

```diff
@@ -0,0 +1,92 @@
+using System.Text.RegularExpressions;
+using FluentValidation;
+using GovUk.Education.ExploreEducationStatistics.Common.Extensions;
+using GovUk.Education.ExploreEducationStatistics.Common.Validators.ErrorDetails;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Model;
+
+namespace GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Validators;
+
+public static partial class SortStringValidators
+{
+    private const string ExpectedFormat = "{field}|{order}";
+    private const int MaxFieldLength = 40;
+
+    public static IRuleBuilderOptionsConditions<T, string> SortString<T>(this IRuleBuilder<T, string> rule)
+    {
+        return rule.Custom((value, context) =>
+        {
+            if (!HasValidFormat(value))
+            {
+                context.AddFailure(
+                    message: ValidationMessages.SortFormat,
+                    detail: new FormatErrorDetail(value, ExpectedFormat: ExpectedFormat)
+                );
+
+                return;
+            }
+
+            var sort = new ParsedSort(value);
+
+            if (!HasValidFieldLength(sort))
+            {
+                context.AddFailure(
+                    message: ValidationMessages.SortMaxFieldLength,
+                    detail: new MaxFieldLengthErrorDetail(sort.String),
+                    new Dictionary<string, object>
+                    {
+                        { "MaxLength", MaxFieldLength }
+                    }
+                );
+            }
+
+            if (!HasValidDirection(sort))
+            {
+                context.AddFailure(
+                    message: ValidationMessages.SortDirection,
+                    detail: new DirectionErrorDetail(sort.String)
+                );
+            }
+        });
+    }
+
+    [GeneratedRegex(@"^[\w_]+\|[A-Za-z]+$", RegexOptions.Compiled, matchTimeoutMilliseconds: 200)]
```

Comment:
After re-reviewing this I'm guessing we need the underscore because we have filters and things that have labels with underscores?

Still not sure what kind of things we would be sorting that has numbers in the name though?

Reply:
Yep pretty much as you've said. Some examples of possible filters that these rules apply for:

- `school_type`
- `ssa_tier_1`, `ssa_tier_2`, etc

So numbers and underscores are needed in the regex.

Reply:
Perfect Regex in that case! :)